validator/client/attest.go

```diff
@@ -237,12 +243,24 @@ func (v *validator) waitToSlotOneThird(ctx context.Context, slot uint64) {
 	}
 	t := time.NewTimer(wait)
 	defer t.Stop()
-	select {
-	case <-ctx.Done():
-		traceutil.AnnotateError(span, ctx.Err())
-		return
-	case <-t.C:
-		return
+
+	bChannel := make(chan *ethpb.SignedBeaconBlock, 1)
+	sub := v.blockFeed.Subscribe(bChannel)
+	defer sub.Unsubscribe()
+
+	for {
+		select {
+		case b := <-bChannel:
+			if slot == b.Block.Slot {
```

Comment:
```suggestion
			if slot <= b.Block.Slot {
```
Some here too

Reply:
I am a bit hesitant about this due to fork. Example:
```
b100 - b102
   \-----b101
```
Say node processes blocks in the order of block at slot 100, 102 and 101, if the attesters at slot 101 request attestation in time before the arrival of 102, the attestation will use 102 as head, instead the attesters should wait until it processes block at 101 and determine whether to use 102 or 101 as head

Reply:
The more I think about this, I think you are right. Beacon node cannot see block 102 before 8 seconds into slot block 101 so the timer would have expired either way

Reply:
I think this situation does not happen: by our clock if we have synced b102 it means it passed the check in `getBlockPreState` that verifies that the slot time is not from the future. It means that  the 4 seconds into b101 have already passed so anyway the timer would have fired and we would have used b100 correctly as head. I see no harm in using `<=` as @prestonvanloon suggest, just that I think it's equivalent to `==`